### PR TITLE
Add new unified memory framework component to oneAPI DPC++ repack

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
## Checklist:

* [x] I want to add a package output to a feedstock:
  * [x] Pinged the relevant feedstock team(s)
  * [x] Added a small description of why the output is being added.

---

@conda-forge/intel-compiler-repack 

This PR adds Unified-Memory-Framework conda package to the allowed outputs of intel-compiler-repack-feedstock to unblock update to 2025.0 version in https://github.com/conda-forge/intel-compiler-repack-feedstock/pull/52

### CODE REVIEW

File: requests/intel-compiler-repack-add-umf.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+action: add_feedstock_output
+feedstock_to_output_mapping:
+  # this entry adds a package name
+  - intel-compiler-repack: "umf-*"
```
Comment (beckermr) [inline requests/intel-compiler-repack-add-umf.yml:4]: Is there a reason you need the entire `umf-*` namespace?

Comment (oleksandr-pavlyk) [reply requests/intel-compiler-repack-add-umf.yml:4]: @beckermr I am not exactly sure. I was trying to accomplish getting a `true` here:  `"linux-64/umf-2025.0.0-h7a4b287_590.conda": false`.   I thought, to accommodate future versions, I need `umf-*`, but I am happy to stand corrected

Comment (beckermr) [reply requests/intel-compiler-repack-add-umf.yml:4]: Ah ok. It is just the package name, so you only need `umf`.

Comment (beckermr) [inline requests/intel-compiler-repack-add-umf.yml:4]:
```suggestion
  - intel-compiler-repack: umf
```

**Pull request merged**